Review and retrospect for sprint 5 (#45)
chhoumann/p5: sections/Chapter 7 Sprint 5/retrospective.tex

\section{Retrospective}

> We have received three user stories from the PO, but have not deployed any of them to the \knox{} pipeline. Thus, we have not completed the sprint goal.

Reviewer: ```suggestion
We have received three user stories from the PO, but have not deployed functionality that satisfies any of them to the \knox{} pipeline. Thus, we have not completed the sprint goal.
```

\subsection{Sprint goal}
The goal of this sprint was to provide endpoints for an RDF database, such that the \knox{} pipeline could fetch data using a SPARQL query, and input data in turtle format.
Since none of these features are deployed, we do not meet the acceptance criteria (see section \ref{acceptCriteriaSprint5}).

\subsection{Reflection on the sprint}
During the sprint, we were made aware that the report needed more theory on databases. 
Therefore some of the tasks regarding report writing grew to an almost unmanageable size. This in turn meant that more time was spend on writing than was originally planned for.
The decision to switch from Fuseki to Virtuoso happened early enough in the sprint that the impact was fairly minor. The time spend researching how to use Fuseki is however wasted due to it no longer being needed.

> During the first sprints of the project our PO did not give us any user stories (epics). Having user the stories for this sprint greatly improved our focus when grooming the backlog, and also help give us a concrete idea of when a tasks was done. 

Reviewer: ```suggestion
During the first sprints of the project, our PO did not give us any user stories (epics). Having user stories for this sprint greatly improved our focus when grooming the backlog, and also help give us a concrete idea of when a task was done. 
```

> Having The user stories for the sprint planning allowed us to granulate them with the other involved layers, which in turn allowed us to remove an unnecessary task. Originally the PO did not want the endpoints to take SPARQL requests as their input.

Reviewer: ```suggestion
The user stories for the sprint planning allowed us to granulate them with the other involved layers, which in turn allowed us to remove an unnecessary task. Originally the PO did not want the endpoints to take SPARQL requests as their input.
```

Author: I don't feel like the sentence makes sence anymore with Having removed?

> The knowledge layer, who needed the endpoints, disagreed, and after a conversation with the PO, it was established that a SPARQL endpoint would be used.

Reviewer: ```suggestion
The knowledge layer, who needed the endpoints, disagreed, and after a conversation with the PO, it was established that endpoints receiving SPARQL queries as a string would be sufficient.
```
